### ISSUE TITLE
Registry NS, NSE connect leaking clients

### ISSUE THREAD

**Issue by Bolodya1997** [networkservicemesh/sdk#777]
# Issue
Expired clients in NS, NSE are getting removed from `cache` but not getting closed:
```go
func (c *connectNSEServer) connect(ctx context.Context) registry.NetworkServiceEndpointRegistryClient {
	key := ""
	if url := clienturlctx.ClientURL(ctx); url != nil {
		key = url.String()
	}
	if v, ok := c.cache.Load(key); v != nil && ok {
		if v.expirationTimer.Stop() {
			v.expirationTimer.Reset(c.connectExpiration)
		}
		return v.client
	}
	ctx = extend.WithValuesFromContext(c.ctx, ctx)
	client := clienturl.NewNetworkServiceEndpointRegistryClient(ctx, c.clientFactory, c.dialOptions...)
	cached, _ := c.cache.LoadOrStore(key, &nseCacheEntry{
		expirationTimer: time.AfterFunc(c.connectExpiration, func() {
			c.cache.Delete(key)
		}),
		client: client,
	})
	return cached.client
}
```
Such case is not covered by unit tests.

**Comment by denis-tingaikin** [comment]
This bug has been added in https://github.com/networkservicemesh/sdk/commit/f97b0e8202783ee9e24ac858ce414659f443a495

**Comment by denis-tingaikin** [comment]
Previously we have used Finalazers for this...

**Comment by Bolodya1997** [comment]
Fixed in #937.
